### PR TITLE
feat(sidebar): add SidebarHeader component

### PR DESCRIPTION
**Pull request by TheSisb**
- [x] Adds the SidebarHeader component
- [x] Tests
- [x] Stories
- [x] Customization

### CODE REVIEW

File: packages/paste-core/components/sidebar/src/SidebarHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        display="flex"
+        alignItems="center"
+        justifyContent="flex-start"
+        columnGap="space50"
+        overflow="hidden"
+        element={element}
+        ref={ref}
+      >
+        <Button size="icon" variant="inverse" element={`${element}_BUTTON`} onClick={onClick}>
+          {productIcon ? productIcon : <ProductHomeIcon size="sizeIcon20" decorative element={`${element}_ICON`} />}
```
Comment (nkrantz) [inline packages/paste-core/components/sidebar/src/SidebarHeader.tsx:35]: suggestion: specify icon color

File: packages/paste-core/components/sidebar/src/SidebarHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        </Button>
+        <Box
+          opacity={collapsed ? '0' : '1'}
+          element={`${element}_TEXT`}
+          fontSize="fontSize30"
+          fontWeight="fontWeightSemibold"
+          lineHeight="lineHeight30"
+          letterSpacing="-.02em"
+          as="h5"
```
Comment (nkrantz) [inline packages/paste-core/components/sidebar/src/SidebarHeader.tsx:44]: question: why did you go with h5

File: packages/paste-core/components/sidebar/src/SidebarCollapseButton.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    return (
+      <Button
+        ref={ref}
+        size="icon"
+        variant="inverse"
+        element={element}
+        {...props}
+        aria-expanded={!collapsed}
+        aria-controls={sidebarId}
```
Comment (SiTaggart) [inline packages/paste-core/components/sidebar/src/SidebarCollapseButton.tsx:32]: praise: excellent inclusion of accessibility

File: packages/paste-core/components/sidebar/src/SidebarPushContentWrapper.tsx [hunk 4]
```diff
@@ -48,4 +48,4 @@ export const PushSidebarContentWrapper = React.forwardRef<HTMLDivElement, PushSi
     );
   }
 );
-PushSidebarContentWrapper.displayName = 'PushSidebarContentWrapper';
+SidebarPushContentWrapper.displayName = 'SidebarPushContentWrapper';
```
Comment (SiTaggart) [inline packages/paste-core/components/sidebar/src/SidebarPushContentWrapper.tsx:51]: praise: ah yeah, these seem like really obvious names now.

File: packages/paste-core/components/sidebar/src/SidebarHeader.tsx (hunk 2 above)
Comment (SiTaggart) [reply packages/paste-core/components/sidebar/src/SidebarHeader.tsx:44]: yeah, I might have some caution on the use of ay heading level given document hierarchy will be unknown. We might just get away with a div here

File: packages/paste-core/components/sidebar/src/SidebarHeader.tsx (hunk 1 above)
Comment (TheSisb) [reply packages/paste-core/components/sidebar/src/SidebarHeader.tsx:35]: I think `currentColor` as default should work fine. Any reason why you recommend specifying it?

**Review by nkrantz — APPROVED** [review]

File: packages/paste-core/components/sidebar/src/SidebarHeader.tsx (hunk 1 above)
Comment (serifluous) [reply packages/paste-core/components/sidebar/src/SidebarHeader.tsx:35]: Does that allow a product to change the icon color? Like, say in the future if we do want each product to use the brand colors for the icon color?

Comment (TheSisb) [reply packages/paste-core/components/sidebar/src/SidebarHeader.tsx:35]: Yup. Teams can pass in their own icon entirely with the `productIcon` prop.

**Pull request merged**